libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```diff
@@ -1,99 +1,226 @@
+      }),
+    )
+
+    const [
+      incomingDelegationSets,
+      outgoingDelegations,
+      generalOutgoingDelegations,
+    ] = await Promise.all([
+      await Promise.all(incomingDelegationPromises),
+      this.delegationModel.findAll({
+        where: {
+          fromNationalId: nationalId,
+        },
+        include: [
+          {
+            model: DelegationScope,
+            required: true,
+            include: [
+              {
+                model: ApiScope,
+                required: true,
+                as: 'apiScope',
+                where: {
+                  enabled: true,
                 },
-              ],
+                include: [
+                  {
+                    model: ApiScopeDelegationType,
+                    required: true,
+                    where: {
+                      delegationType: AuthDelegationType.Custom,
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      }),
+      this.delegationModel.findAll({
+        where: {
+          fromNationalId: nationalId,
+        },
+        include: [
+          {
+            model: DelegationDelegationType,
+            required: true,
+            where: {
+              delegationTypeId: AuthDelegationType.GeneralMandate,
             },
-          ],
+          },
+        ],
+      }),
+    ])
+
+    return {
+      incoming: ([] as DelegationDTO[]).concat(...incomingDelegationSets),
+      outgoing: [
+        generalOutgoingDelegations.map((d) =>
+          d.toDTO(AuthDelegationType.GeneralMandate),
+        ),
+        outgoingDelegations.map((delegation) => delegation.toDTO()),
+      ].flat(),
+    }
+  }
+
+  async createDelegation(
+    user: User,
+    delegation: CreatePaperDelegationDto,
+  ): Promise<DelegationDTO> {
+    this.validatePersonsNationalIds(
+      delegation.toNationalId,
+      delegation.fromNationalId,
+    )
+
+    const zenDeskCase = await this.zendeskService.getTicket(
+      delegation.referenceId,
+    )
+
+    if (!zenDeskCase.tags.includes(DELEGATION_TAG)) {
+      throw new Error('Zendesk ticket is missing required tag')
+    }
+
+    if (zenDeskCase.status !== TicketStatus.Solved) {
+      throw new Error('Zendesk case is not solved')
+    }
+
+    const { fromReferenceId, toReferenceId } =
+      this.getNationalIdsFromZendeskTicket(zenDeskCase)
+
+    if (
+      fromReferenceId !== delegation.fromNationalId ||
+      toReferenceId !== delegation.toNationalId
+    ) {
+      throw new Error(
+        'Zendesk ticket nationalIds does not match delegation nationalIds',
+      )
+    }
+
+    const [fromDisplayName, toName] = await Promise.all([
+      this.namesService.getPersonName(delegation.fromNationalId),
+      this.namesService.getPersonName(delegation.toNationalId),
+    ])
+
+    return this.sequelize.transaction(async (transaction) => {
+      const newDelegation = await this.delegationModel.create(
```

Comment:
With Sequelize you can create the delegation with the DelegationType association in one step

https://sequelize.org/docs/v6/advanced-association-concepts/creating-with-associations/